src/variant_detection/variant_output.cpp

```diff
@@ -66,58 +66,56 @@ void write_record(Cluster const & cluster,
     record.filter() = {"PASS"};
     record.genotypes() = {};
     record.genotypes().push_back({ .id = "GT", .value = std::vector{"./."s}});
+    record.info() = {};
 
     if (mate1.orientation == mate2.orientation)
     {
         if (mate1.seq_name == mate2.seq_name)
         {
-            size_t insert_size = cluster.get_average_inserted_sequence_size();
             if (mate1.orientation == strand::forward)
             {
+                size_t insert_size = cluster.get_average_inserted_sequence_size();
                 int distance = mate2.position - mate1.position - 1;
-                int sv_length = insert_size - distance;
+                int sv_length;
+                std::string sv_type;
+                // Tandem Duplication
+                if (cluster.get_common_tandem_dup_count() > 0)
+                {
+                    record.alt() = {"<DUP:TANDEM>"};
+                    // Increment end by 1 because VCF is 1-based
+                    record.info().push_back({.id = "END", .value = mate2.position + 1});
+                    sv_length = insert_size;
+                    sv_type = "DUP";
+                }
+                // Deletion
+                else if (distance > 0)
+                {
+                    record.alt() = {"<DEL>"};
+                    // Increment end by 1 because VCF is 1-based
+                    // Decrement end by 1 because deletion ends one base before mate2 begins
+                    record.info().push_back({.id = "END", .value = mate2.position});
+                    sv_length = -distance;
+                    sv_type = "DEL";
+                }
+                // Insertion (sv_length is positive)
+                // for a small deletion inside of an insertion, the distance is a small negative value
+                else if (insert_size > 0 && distance <= 0)
+                {
+                    record.alt() = {"<INS>"};
+                    // Increment end by 1 because VCF is 1-based
+                    record.info().push_back({.id = "END", .value = mate1.position + 1});
+                    sv_length = insert_size;
+                    sv_type = "INS";
+                }
                 // The SVLEN is neither too short nor too long than specified by the user.
                 if (std::abs(sv_length) >= args.min_var_length &&
-                    std::abs(sv_length) <= args.max_var_length)
+                    std::abs(sv_length) <= args.max_var_length &&
+                    ((sv_type != "DUP" || sv_type != "INS") || insert_size <= args.max_tol_deleted_length) &&
```

Comment:
I don't understand this condition, and `(sv_type != "DUP" || sv_type != "INS")` is always true.

Also note that our definition of SVLEN has impact here.

Reply:
Ah damn my summary has ruined it 😅 thx for the hint